device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java

```diff
@@ -419,7 +419,60 @@ public void open() throws TransportException
 
         this.taskScheduler = Executors.newScheduledThreadPool(1);
 
-        openConnection();
+        if (withRetry)
+        {
+            int connectionAttempt = 0;
+            long startTime = System.currentTimeMillis();
+
+            // this loop either ends in throwing an exception when retry expires, or by a break statement upon a successful openConnection() call
+            while (true)
+            {
+                RetryPolicy retryPolicy;
+                if (isMultiplexing)
```

Comment:
does Java not support ternary operator?

Reply:
It does, and I can make this a single line, sure